CoastalResilienceNetwork/GeositeFramework
# Auto-resize legend

Only 0eea00e is part of this PR. Also, I'm still testing out edge cases.

* Resize the legend as layers are turned on so that the legend box grows to fit the additional legends without needing a scroll bar. This continues until the legend hits hard height and width limits. Then a scroll bar is added.

**Testing instructions**
- Add layers and ensure the legend resizes to display all the layers until it reaches `MAX_HEIGHT` and `MAX_WIDTH`.  A scroll bar should be added then.
- Manually make the legend really large. Add and remove layers. It should not resize. The idea here is that if the user manually resized the legend, we shouldn't mess with it.
- Make the legend small again, and add/remove layers. It should resize.

![legend](https://cloud.githubusercontent.com/assets/1042475/12454333/01781f1a-bf65-11e5-85e4-3f17cccd1cdf.gif)

Connects to #494
